### PR TITLE
chore: New Auth for the Projects

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Part of:
https://github.com/webstudio-is/webstudio-saas/issues/317#issuecomment-2289241819

This configuration is necessary for the new authentication process.

- NODE_TLS_REJECT_UNAUTHORIZED: Enables service-to-service communications by allowing self-signed certificates.
- cors: Ensures CORS settings match those in the production environment.
- cors logout: Allows logout requests on subdomains using fetch with credentials.

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

File: apps/builder/app/shared/router-utils/origins.server.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  const { projectId } = parseBuilderUrl(urlStr);
+  return projectId !== undefined;
+};
+
+export function getAuthorizationServerOrigin(request: Request): string;
+export function getAuthorizationServerOrigin(origin: string): string;
+
+// eslint-disable-next-line func-style
+export function getAuthorizationServerOrigin(
+  request: string | Request
```
Comment (TrySound) [inline apps/builder/app/shared/router-utils/origins.server.ts:73]: why overloading is needed here?

Comment (istarkov) [reply apps/builder/app/shared/router-utils/origins.server.ts:73]: Its just 1st PR of many, its needed

Comment (istarkov) [reply apps/builder/app/shared/router-utils/origins.server.ts:73]: See
export const getRequestOrigin = (request: Request) => {
  const url = new URL(request.url);

  // vercel overwrites x-forwarded-host on edge level even if our header is set
  // as workaround we use custom header x-forwarded-ws-host to get the original host
  url.host =
    request.headers.get('x-forwarded-ws-host') ??
    request.headers.get('x-forwarded-host') ??
    url.host;
  url.protocol = request.headers.get('x-forwarded-proto') ?? 'https';
  return url.origin;
};

Comment (istarkov) [reply apps/builder/app/shared/router-utils/origins.server.ts:73]: In some cases I have only URL in some it's known only through a headers.

**Review by TrySound — APPROVED** [review]

**Pull request merged**